### PR TITLE
 Use WaitForSynced in validator client for startup

### LINKED ISSUE #5452: Misleading log message from validator before genesis start
I see "Waiting for beacon node to sync to latest chain head" even the chain is in sync with head before genesis. The message is misleading. It would be changed or improved.

```
time="2020-04-16 05:48:18" level=info msg="Beacon chain started" genesisTime=2020-04-18 02:00:00 +0200 CEST prefix=validator
--
  | time="2020-04-16 05:48:24" level=info msg="Waiting for beacon node to sync to latest chain head" prefix=validator
  | time="2020-04-16 05:48:30" level=info msg="Waiting for beacon node to sync to latest chain head" prefix=validator
  | time="2020-04-16 05:48:36" level=info msg="Waiting for beacon node to sync to latest chain head" prefix=validator
  | time="2020-04-16 05:48:42" level=info msg="Waiting for beacon node to sync to latest chain head" prefix=validator
  | time="2020-04-16 05:48:48" level=info msg="Waiting for beacon node to sync to latest chain head" prefix=validator
```

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #5345, meant to be merged after #5366 , this PR changes the validator client to use `WaitForSynced` to listen for what the validator is ready, rather than pinging if the node is synced every 6 seconds. 

This feature is gated by the `--wait-for-synced` flag on the validator client.
Confirmed in runtime:
Old Startup: 
```
time="2020-04-16 19:40:42" msg="Beacon node is ready 2.852288s after chainstart"
time="2020-04-16 19:40:42" msg="Activations received 2.853108s after chainstart"
```
New Startup: 
```
time="2020-04-16 19:48:10" msg="Beacon node is ready 45.511ms after chain start"
time="2020-04-16 19:48:10" msg="Activations received 78.148ms after chainstart"
```
Resolves #5452

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Tests fail. Please wait to merge new features until after Topaz starts

**Comment by nisdas** [comment]
Is this unblocked now @0xKiwi ?

**Comment by 0xKiwi** [comment]
Yes @nisdas

**Comment by 0xKiwi** [comment]
I did! This is the validator half but its waiting on #5366 to be merged first so it has changes for both :)

**Comment by nisdas** [comment]
Ah #5366 is meant to be merged first, just saw it. Lets merge that then

File: validator/client/runner.go [hunk 1]
```diff
@@ -45,11 +46,8 @@ type Validator interface {
 // 6 - Perform assigned role, if any
 func run(ctx context.Context, v Validator) {
 	defer v.Done()
-	if err := v.WaitForChainStart(ctx); err != nil {
```
Comment (0xKiwi) [inline validator/client/runner.go:48]: Should I be removing these functions from the validator client in this PR or should we do that later?

File: beacon-chain/sync/initial-sync/service.go [hunk 2]
```diff
@@ -178,6 +178,7 @@ func (s *Service) Syncing() bool {
 func (s *Service) Resync() error {
 	// set it to false since we are syncing again
 	s.synced = false
+	// How does the validator handle this?
```
Comment (0xKiwi) [inline beacon-chain/sync/initial-sync/service.go:181]: Should there be anything special done here? Missed it in the beacon-chain PR.

Comment (nisdas) [reply beacon-chain/sync/initial-sync/service.go:181]: Don't think you can, you would have to create new methods in the validator client to handle this.

File: validator/client/validator.go [hunk 3]
```diff
@@ -146,6 +146,40 @@ func (v *validator) WaitForSync(ctx context.Context) error {
...
+// fully synced and ready to communicate with the validator.
+func (v *validator) WaitForSynced(ctx context.Context) error {
+	ctx, span := trace.StartSpan(ctx, "validator.WaitForSynced")
+	defer span.End()
+	// First, check if the beacon chain has started.
+	stream, err := v.validatorClient.WaitForSynced(ctx, &ptypes.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "could not setup beacon chain Synced streaming client")
+	}
+	for {
```
Comment (shayzluf) [inline validator/client/validator.go:159]: is there a need for this for loop. seem to happen once

**Pull request merged**